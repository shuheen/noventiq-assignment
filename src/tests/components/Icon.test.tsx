// Components
import Icon from '../../components/Icons/Icon';
import { render } from '@testing-library/react';

const clickMock = jest.fn();
// Tests
describe('Tests for <Icon /> component', () => {
  it('Renders with default props', () => {
    const { container } = render(<Icon name="password" />);
    // Create the snapshot test
    expect(container).toMatchSnapshot();
  });
});
