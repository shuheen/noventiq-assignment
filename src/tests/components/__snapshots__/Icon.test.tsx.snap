// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Icon /> component Renders with default props 1`] = `
<div>
  <svg
    class=""
    data-testid="icon-password"
    role="icon"
    style="width: 48px; height: 48px;"
    viewBox="0 0 48 48"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15.8683 16C15.9121 16 15.956 16 16 16L32.1317 16C33.9046 15.9999 35.4323 15.9998 36.6556 16.1643C37.9666 16.3406 39.2233 16.738 40.2427 17.7574C41.262 18.7767 41.6595 20.0334 41.8357 21.3444C42.0002 22.5678 42.0001 24.0954 42 25.8684L42 30.1411C42.0001 32.8494 42.0001 35.0826 41.7625 36.8501C41.513 38.7058 40.9691 40.3446 39.6569 41.6569C38.3446 42.9691 36.7058 43.513 34.8501 43.7625C33.0826 44.0001 30.8494 44.0001 28.1412 44H19.8589C17.1506 44.0001 14.9174 44.0001 13.1499 43.7625C11.2942 43.513 9.65537 42.9691 8.34316 41.6569C7.03094 40.3446 6.487 38.7058 6.23751 36.8501C5.99988 35.0826 5.99994 32.8494 6.00001 30.1412L6.00001 26C6.00001 25.956 6.00001 25.9121 6 25.8683C5.99991 24.0954 5.99982 22.5677 6.16429 21.3444C6.34055 20.0334 6.73804 18.7767 7.75737 17.7574C8.77669 16.738 10.0334 16.3406 11.3444 16.1643C12.5677 15.9998 14.0954 15.9999 15.8683 16ZM11.8774 20.1286C11.0099 20.2453 10.738 20.4335 10.5858 20.5858C10.4335 20.738 10.2453 21.0099 10.1286 21.8774C10.0043 22.8024 10 24.0579 10 26V30C10 32.885 10.0043 34.8475 10.2018 36.3171C10.3917 37.7293 10.7264 38.3833 11.1716 38.8284C11.6167 39.2736 12.2708 39.6083 13.6829 39.7982C15.1525 39.9958 17.115 40 20 40H28C30.885 40 32.8475 39.9958 34.3171 39.7982C35.7293 39.6083 36.3833 39.2736 36.8284 38.8284C37.2736 38.3833 37.6083 37.7293 37.7982 36.3171C37.9958 34.8475 38 32.885 38 30V26C38 24.0579 37.9958 22.8024 37.8714 21.8774C37.7548 21.0099 37.5665 20.738 37.4142 20.5858C37.262 20.4335 36.9901 20.2453 36.1226 20.1286C35.1976 20.0043 33.9422 20 32 20H16C14.0579 20 12.8024 20.0043 11.8774 20.1286Z"
      fill-rule="evenodd"
      style="fill: currentColor;"
    />
    <path
      d="M14 14C14 8.47715 18.4772 4 24 4C29.5228 4 34 8.47715 34 14V16C34 17.1046 33.1046 18 32 18C30.8954 18 30 17.1046 30 16V14C30 10.6863 27.3137 8 24 8C20.6863 8 18 10.6863 18 14V16C18 17.1046 17.1046 18 16 18C14.8954 18 14 17.1046 14 16V14Z"
      fill-rule="evenodd"
      style="fill: currentColor;"
    />
  </svg>
</div>
`;
